src/Domain/FunctionsOfTime/QuaternionHelpers.hpp

```diff
@@ -27,21 +23,9 @@ DataVector quaternion_to_datavector(
 /// 0 scalar part while the vector part is the `DataVector`. If the `DataVector`
 /// has 4 components, the quaternion is just the `DataVector` itself.
 boost::math::quaternion<double> datavector_to_quaternion(
-    const DataVector& input) noexcept {
-  ASSERT(input.size() == 3 or input.size() == 4,
-         "To form a quaternion, a DataVector can either have 3 or 4 components "
-         "only. This DataVector has "
-             << input.size() << " components.");
-  if (input.size() == 3) {
-    return boost::math::quaternion<double>(0.0, input[0], input[1], input[2]);
-  } else {
-    return boost::math::quaternion<double>(input[0], input[1], input[2],
-                                           input[3]);
-  }
-}
+    const DataVector& input) noexcept;
+
 /// Normalize a `boost::math::quaternion`
 template <typename T>
 void normalize_quaternion(
-    const gsl::not_null<boost::math::quaternion<T>*> input) noexcept {
-  *input /= abs(*input);
-}
+    const gsl::not_null<boost::math::quaternion<T>*> input) noexcept;
```

Comment:
Not sure why clang-tidy doesn't complain here, but this declaration shouldn't have the `const` for the argument variable, only the definition in the cpp needs it. Note that the `const` here applies to the variable; it is _not_ part of the argument type as it is for const-ref arguments like `const DataVector&`.